src/test/lib/utils/print_utils_test.cpp

```diff
@@ -1,13 +1,65 @@
 #include "base_test.hpp"
 
+#include "logical_query_plan/mock_node.hpp"
 #include "storage/table.hpp"
 #include "utils/print_utils.hpp"
 
 namespace opossum {
 
 class PrintUtilsTest : public BaseTest {};
 
-TEST_F(PrintUtilsTest, table_key_constraints_to_stream) {
+TEST_F(PrintUtilsTest, print_directed_acyclic_graph) {
+  const auto column_definitions = MockNode::ColumnDefinitions{};
+  const auto recurring_node = MockNode::make(column_definitions, "recurring_node");
+  // clang-format off
+  const auto lqp =
+  MockNode::make(column_definitions, "node_a",
+    MockNode::make(column_definitions, "node_b",
+      recurring_node),
+    MockNode::make(column_definitions, "node_c",
+      recurring_node,
+      recurring_node));
+  // clang-format on
+
+  // Functor to access the MockNode's inputs
+  const auto get_inputs_fn = [](const auto& node) {
+    std::vector<std::shared_ptr<const AbstractLQPNode>> inputs;
+    if (node->left_input()) {
+      inputs.emplace_back(node->left_input());
+    }
+
+    if (node->right_input()) {
+      inputs.emplace_back(node->right_input());
+    }
+
+    return inputs;
+  };
+
+  // Functor to print the MockNode's name
+  const auto node_print_fn = [](const auto& node, auto& stream) {
+    const auto& mock_node = static_cast<const MockNode&>(*node);
+    stream << *mock_node.name;
+  };
+
+  auto stream = std::ostringstream{};
+
+  /**
+   * [0] node_a
+   *  \_[1] node_b
+   *  |  \_[2] recurring_node
+   *  \_[3] node_c
+   *     \_Recurring Node --> [2]
+   *     \_Recurring Node --> [2]
+   */
+  const auto expected_string = std::string{
+      "[0] node_a\n \\_[1] node_b\n |  \\_[2] recurring_node\n \\_[3] node_c\n    \\_Recurring Node --> [2]\n    "
+      "\\_Recurring Node --> [2]\n"};
```

Comment:
Instead of adding the pretty formatted string representation above as a comment and the expected string in a compact format, how about removing the comment and formatting the expected string similar to the current comment? Then, we would not have the string twice and one could argue that this would also increase the expected string's readablity.

Reply:
I initially dit it this way, but escaping the backslashes (adds one character offset at some places) and adding newline characters in the string kind of did not made this look nice IMO:
```c++
const auto expected_string = std::string{
  "[0] node_a\n"
  " \\_[1] node_b\n"
  " |  \\_[2] recurring_node\n"
  " \\_[3] node_c\n"
  "    \\_Recurring Node --> [2]\n"
  "    \\_Recurring Node --> [2]\n"};
```

Reply:
Even with the escaping backslashes and the `\n`s, I think the structure of the lqp is quite clear with that formatting.
Pinging @Bouncner and @tjjordan for more opinions on that.

Reply:
Hm, in my view both options are totally fine.
Wouldn't it be an option to keep both the nice formatted comment and the nice formatted ´expected_string`?
If we want to get rid of the duplicate I'd prefer the nice formatted ´expected_string` because it makes more clear how the string should be build.

Reply:
Current version is:
```c++
  auto stream = std::ostringstream{};
  const auto expected_string = std::string{
      R"([0] node_a
 \_[1] node_b
 |  \_[2] recurring_node
 \_[3] node_c
    \_Recurring Node --> [2]
    \_Recurring Node --> [2]
)"};

  print_directed_acyclic_graph<const AbstractLQPNode>(lqp, get_inputs_fn, node_print_fn, stream);
  EXPECT_EQ(stream.str(), expected_string);
```